src/commands/login.ts

```diff
@@ -20,23 +21,40 @@ export default class Login extends Command {
       const { auth } = config
       try {
         await api.checkAuth(auth)
-        this.log(`User ${config.username} is already logged in. To log in with a different user, run jamsocket logout first.`)
+        this.log('You are already logged in. To log in with a different token, run jamsocket logout first.')
         return
       } catch (error) {
         const isAuthError = error instanceof AuthenticationError
         if (!isAuthError) throw error
       }
     }
 
-    const username = await CliUx.ux.prompt('username')
-    const password = await CliUx.ux.prompt('password', { type: 'hide' })
+    const { args } = await this.parse(Login)
+
+    let token = args.token.trim()
+    if (!token) {
+      this.log('Generate an API token at the following URL and paste it into the prompt below:\n')
+      this.log(`    ${api.getLoginUrl()}\n`)
+      token = (await CliUx.ux.prompt('token')).trim()
+    }
+
+    if (!token) {
+      throw new Error('Token required to login.')
+    }
+
+    if (!token.includes('.')) {
+      throw new Error('Invalid token. Token must contain a period.')
+    }
+
+    const [publicPortion, privatePortion] = token.split('.')
 
-    const buff = Buffer.from(`${username}:${password}`, 'utf-8')
+    const buff = Buffer.from(`${publicPortion}:${privatePortion}`, 'utf-8')
     const auth = buff.toString('base64')
 
     await api.checkAuth(auth)
 
-    writeJamsocketConfig({ username: username, auth: auth })
+    // should tokens be stored like this? or should they explicitly be tokens - no username?
+    writeJamsocketConfig({ username: publicPortion, auth: auth })
```

Comment:
The nice thing about this approach is that it just works with no other changes to this codebase. However, it doesn't really reflect what we're actually storing. The API token public portion isn't _really_ a `username`. Maybe it's better to store tokens in a `token` field, and then have them sent to the REST API with Bearer Auth and have them encoded as `base64(public:private)` for docker auth. That might make more sense, but requires more changes. Thoughts?

Reply:
I agree but it's not urgent, I'm fine with this for now.